### ISSUE TITLE
[EKS] [request]: Enable code pipeline support for private endpoints

### ISSUE THREAD

**Issue by abhishekjawali** [aws/containers-roadmap#574]
**Tell us about your request**
Use Codepipeline to create EKS with private endpoint and also to deploy apps to EKS with private endpoint

**Which service(s) is this request for?**
EKS, Codepipeline

**Tell us about the problem you're trying to solve. What are you trying to do, and why is it hard?**
When we enable "private endpoint access" in EKS cluser, then this cluster can only be accessed from the VPC it is tagged to. AWS recommends to use bastion host for this. But this defeats automation and lot of workaround is required to deploy apps to EKS using bastion host. 

Codepipeline gives required automation to create and deploy apps to EKS with public endpoint. However, when we have only private endpoint, Codepipeline is not able to connect to EKS cluster. 

I feel if we have Codepipeline access the EKS cluster with private endpoint, then we can have a clean approach to create EKS and deploy apps to EKS.

**Are you currently working around this issue?**
Yes. By having a Bastion host within VPC.

**Comment by tabern** [comment]
@abhishekjawali would Security Groups on the public endpoint help with this? https://github.com/aws/containers-roadmap/issues/108

**Comment by abhishekjawali** [comment]
@tabern , yes, Security groups on public endpoints will definitely help here to an extent. This will help in restricting access to public endpoint to known sources.
